librz/asm/arch/i4004/i4004dis.c

```diff
@@ -62,40 +62,86 @@ static int i4004dis(RzAsmOp *op, const ut8 *buf, int len) {
 	int rlen = i4004_get_ins_len(*buf);
 	ut8 high = (*buf & 0xf0) >> 4;
 	ut8 low = (*buf & 0xf);
-	const char *buf_asm = "invalid";
+	char *buf_asm = rz_str_newf("invalid");
```

Comment:
same here. add a `default` case, set the invalid there and remove the added `free` but not the one after the `rz_strbuf_set`